python/pyarrow/tests/test_sparse_tensor.py

```diff
@@ -0,0 +1,228 @@
+    data[9, 0] = 1.0
+    sparse_tensor2 = sparse_tensor_type.from_dense_numpy(np.ascontiguousarray(data))
+    ne(sparse_tensor1, sparse_tensor2)
+
+
+@pytest.mark.parametrize('dtype_str,arrow_type', [
+    ('i1', pa.int8()),
+    ('i2', pa.int16()),
+    ('i4', pa.int32()),
+    ('i8', pa.int64()),
+    ('u1', pa.uint8()),
+    ('u2', pa.uint16()),
+    ('u4', pa.uint32()),
+    ('u8', pa.uint64()),
+    ('f2', pa.float16()),
+    ('f4', pa.float32()),
+    ('f8', pa.float64())
+])
+def test_sparse_tensor_coo_from_dense(dtype_str, arrow_type):
+    dtype = np.dtype(dtype_str)
+    data_vector = np.array([4, 9, 7, 5]).astype(dtype)
+    coords = np.array([[0, 0], [1, 3], [0, 2], [1, 3]])
+    data = np.array([[4, 0, 9, 0],
+                     [0, 7, 0, 0],
+                     [0, 0, 0, 0],
+                     [0, 0, 0, 5]]).astype(dtype)
+    tensor = pa.Tensor.from_numpy(data)
+
+    # Test from numpy array
+    sparse_tensor = pa.SparseTensorCOO.from_dense_numpy(data)
+    repr(sparse_tensor)
+    assert sparse_tensor.type == arrow_type
+    result_data, result_coords = sparse_tensor.to_numpy()
+    assert (data_vector == result_data).all()
+    assert (result_coords == coords).all()
+
+    # Test from Tensor
+    sparse_tensor = pa.SparseTensorCOO.from_tensor(tensor)
+    repr(sparse_tensor)
+    assert sparse_tensor.type == arrow_type
+    result_data, result_coords = sparse_tensor.to_numpy()
+    assert (data_vector == result_data).all()
+    assert (result_coords == coords).all()
+
+
+@pytest.mark.parametrize('dtype_str,arrow_type', [
+    ('i1', pa.int8()),
+    ('i2', pa.int16()),
+    ('i4', pa.int32()),
+    ('i8', pa.int64()),
+    ('u1', pa.uint8()),
+    ('u2', pa.uint16()),
+    ('u4', pa.uint32()),
+    ('u8', pa.uint64()),
+    ('f2', pa.float16()),
+    ('f4', pa.float32()),
+    ('f8', pa.float64())
+])
+def test_sparse_tensor_csr_from_dense(dtype_str, arrow_type):
+    dtype = np.dtype(dtype_str)
+    dense_data = np.array([[1, 0, 2],
+                           [0, 0, 3],
+                           [4, 5, 6]]).astype(dtype)
+
+    data = np.array([1, 2, 3, 4, 5, 6]).astype(dtype)
+    indptr = np.array([0, 2, 3, 6])
+    indices = np.array([0, 2, 2, 0, 1, 2])
+    tensor = pa.Tensor.from_numpy(dense_data)
+
+    # Test from numpy array
+    sparse_tensor = pa.SparseTensorCSR.from_dense_numpy(dense_data)
+    repr(sparse_tensor)
+    result_data, result_indptr, result_indices = sparse_tensor.to_numpy()
+    assert (data == result_data).all()
+    assert (indptr == result_indptr).all()
+    assert (indices == result_indices).all()
+
+    # Test from Tensor
+    sparse_tensor = pa.SparseTensorCSR.from_tensor(tensor)
+    repr(sparse_tensor)
+    assert sparse_tensor.type == arrow_type
+    result_data, result_indptr, result_indices = sparse_tensor.to_numpy()
+    assert (data == result_data).all()
+    assert (indptr == result_indptr).all()
+    assert (indices == result_indices).all()
+
+
+@pytest.mark.skip
```

Comment:
For some reason this test fails in CI. I can't reproduce it locally.
E.g. [https://travis-ci.org/apache/arrow/jobs/547437221](https://travis-ci.org/apache/arrow/jobs/547437221).
```
pyarrow/tests/test_serialization.py .........ss.......s.............     [ 87%]
pyarrow/tests/test_sparse_tensor.py ......................../Users/travis/build/apache/arrow/ci/travis_script_python.sh: line 201: 14116 Abort trap: 6           (core dumped) python -m pytest $PYARROW_PYTEST_FLAGS pyarrow/tests
The command "$TRAVIS_BUILD_DIR/ci/travis_script_python.sh 3.6" exited with 134.
```

Reply:
Hmm, we should install `pytest-faulthandler` on CI.

Reply:
Also, did you compile in debug mode?